pkg/controller/user-cluster-controller-manager/resources/reconciler.go

```diff
@@ -71,16 +72,24 @@ func (r *reconciler) reconcile(ctx context.Context) error {
 	if err != nil {
 		return fmt.Errorf("failed to get userSSHKeys: %v", err)
 	}
-	cloudConfig, err := r.cloudConfig(ctx)
+	cloudConfig, err := r.cloudConfig(ctx, resources.CloudConfigConfigMapName)
 	if err != nil {
 		return fmt.Errorf("failed to get cloudConfig: %v", err)
 	}
+	var vsphereCSICloudConfig []byte
```

Comment:
Tiny nit: we don't need the `vsphere` prefix, similar to the variable in the `reconcileData` struct.